privacy_evaluator/utils/model_utils.py

```diff
@@ -0,0 +1,67 @@
+from tensorflow import keras
+from torch import nn
+from art.estimators.estimator import BaseEstimator
+import numpy as np
+from copy import deepcopy
+from typing import Union
+import shutil
+
+
+def copy_and_reset_model(
+    model: Union[keras.Model, nn.Module, BaseEstimator]
+) -> Union[nn.Module, keras.Model]:
+    """
+    Get a clone of a model (i.e. with the same architecture) and reset the weights \
+        of this copy.
+
+    :param model: A TF or Torch model. Alternatively, an art-classifier encapsulating \
+        a TF or Torch model.
+    :return: The re-initialized copy of the given model. If `model` is a TF model \
+        (or art-TF) then we return a TF model; the same applied in Torch case.
+    """
+    if isinstance(model, nn.Module):
+        return _copy_and_reset_torch_model(model)
+    if isinstance(model, keras.Model):
```

Comment:
Above you are checking for TensorFlow version 2 model. I would prefer doing this here as well. Not every TensorFlow model is a keras model. But the functionality should also work with other TensorFlow models - unless you throw an error before in the function. 
Or is this case covert with the base estimator? In that case, and maybe in general, I would suggest you adding documentation on what happens within which if case.

Reply:
Here since we have checked the type before, and we add `_` to name the two helper functions, which means we want to make it private and it is not our responsibility if someone else wants to call a private `_copy_and_reset_model` without doing it through the wrapper function `copy_and_reset_model`.
And for `copy_and_reset_model`, if the model type is neither torch nor keras, then we just throw a type error. It is impossible to set warnings exclusive for TF1 `tf.model` case, because `tf.model` does not exist for TF2 version and it is not quite feasible to install two different versions of TF just to get `tf.model` for type checking.